src/reporters/http_sender.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+
+import fs from 'fs';
+import http from 'http';
+import path from 'path';
+import * as url from 'url';
+import { Thrift } from 'thriftrw';
+import NullLogger from '../logger.js';
+import SenderUtils from './sender_utils.js';
+
+const DEFAULT_PATH = '/api/traces';
+const DEFAULT_PORT = 14268;
+const DEFAULT_TIMEOUT_MS = 5000;
+const DEFAULT_MAX_SPAN_BATCH_SIZE = 100;
+
+export default class HTTPSender {
+  _url: Object;
+  _username: string;
+  _password: string;
+  _emitSpanBatchOverhead: number;
+  _timeoutMS: number;
+  _httpAgent: http$Agent;
+  _logger: Logger;
+  _jaegerThrift: Thrift;
+  _process: Process;
+  _batch: Batch;
+  _thriftProcessMessage: any;
+  _maxSpanBatchSize: number;
+
+  constructor(options: any = {}) {
+    this._url = url.parse(
+      `${options.useHTTPS ? 'https' : 'http'}://${options.host || 'localhost'}:${options.port ||
+        DEFAULT_PORT}${options.path || DEFAULT_PATH}`
+    );
+    this._username = options.username;
+    this._password = options.password;
+    this._timeoutMS = options.timeoutMS || DEFAULT_TIMEOUT_MS;
+    this._httpAgent = new http.Agent({ keepAlive: true });
+
+    this._maxSpanBatchSize = options.maxSpanBatchSize || DEFAULT_MAX_SPAN_BATCH_SIZE;
+
+    this._logger = options.logger || new NullLogger();
+    this._jaegerThrift = new Thrift({
+      source: fs.readFileSync(path.join(__dirname, '../jaeger-idl/thrift/jaeger.thrift'), 'ascii'),
+      allowOptionalArguments: true,
+    });
+  }
+
+  setProcess(process: Process): void {
```

Comment:
is this not the same as in the UDP sender? Any reason not to extract into a base class?

Reply:
There is quite a bit of common logic so I'll extract it to the SenderUtils class.